### PR TITLE
Adds eclipse settings files to set correct encoding and complier preferences

### PR DESCRIPTION
**Pull request by colings86**

### CODE REVIEW

**Comment by jpountz** [comment]
Maybe we could try to use the [additionalConfig](https://maven.apache.org/plugins/maven-eclipse-plugin/eclipse-mojo.html#additionalConfig) option in the parent pom to avoid duplicating this option in every module?

File: pom.xml [hunk 1]
```diff
@@ -1034,6 +1034,67 @@
...
+]]>
+								</content>
+							</file>
+							<file>
+								<name>.settings/org.eclipse.m2e.core.prefs</name>
```
Comment (jpountz) [inline pom.xml]: Is it expected to set m2e config through the eclipse plugin?

File: pom.xml [hunk 2]
```diff
@@ -1034,6 +1034,67 @@
...
+							<file>
+								<name>.settings/org.eclipse.core.resources.prefs</name>
+								<content>
+						       <![CDATA[eclipse.preferences.version=1
+encoding//src/main/java=UTF-8
+encoding//src/main/resources=UTF-8
+encoding//src/test/java=UTF-8
+encoding//src/test/resources=UTF-8
+encoding/<project>=UTF-8
+encoding/rest-api-spec=UTF-8
```
Comment (jpountz) [inline pom.xml]: not sure it's relevant anymore now that the rest api spec is only in the parent project?

File: pom.xml [hunk 3]
```diff
@@ -1034,6 +1034,67 @@
                     <configuration>
                         <buildOutputDirectory>eclipse-build</buildOutputDirectory>
                         <downloadSources>true</downloadSources>
+                        <additionalConfig>
+							<file>
```
Comment (jpountz) [inline pom.xml]: Looks like you are using tabs from this line onwards, can you switch to spaces instead?

**Comment by jpountz** [comment]
Left some minor comments, but I like having everything factored out in the parent pom!

**Comment by colings86** [comment]
@jpountz fixed your comments and pushed an update.

**Comment by jpountz** [comment]
LGTM!

**Comment by jpountz** [comment]
@colings86 It seems to me like this change has been pushed already?

**Comment by colings86** [comment]
Yes, it has, not sure why it didn't auto-close

**Pull request closed without merging**